Client/mainwindow.cpp

```diff
@@ -123,8 +124,12 @@ MainWindow::MainWindow(QWidget* parent)
 	// Update server list
 	updateServerList();
 
+	assetManagerTemplate = new ppvs::AssetManager(nullptr, nullptr);
```

Comment:
This variable and all of its surrounding confusion was introduced with to `AssetManager`'s dual job:
1. The manager handles opening bundles and allowing the `Client` to index their contents (in terms of available themes and characters)
2. Each one also has to act as a `Frontend` proxy that verifies the actual contents of available themes, chooses a suitable candidate and presents a simple API for `Puyolib` to use.

An `AssetManager` (`AssetBundleBundle` is not a great name) object is merely a collection of `AssetBundle`s (which themselves can contain a number of themes as well as system assets) implementing greedy overlay, allowing debugging and simplifying the process for the Game.

In the case of Client (old) we create a hollow object containing only the loaded Bundles that we wish to use. The index is currently used primarily by the `SettingsDialog`. The proxy is used by `GameManager` that creates a realization of the template, fills it with Game-specific instances of the debugger and `Frontend` and lets the `Game` use and manage it.

This approach isn't blocking dynamic asset loading as much as `Puyolib`'s inability to withstand rendering unloaded textures and possible race conditions arising.

I suppose I could create something akin to `AssetManagerGenerator` as a member of `MainWindow` that can proxy the needs of both but that sounds rather ridiculous and I am not too sure it would increase readability of `Client`. Splitting each bundle into an index class and a rendering unit child would create two closely interlinked classes that might as well be one.